### ISSUE TITLE
Unhandled favorite

### ISSUE THREAD

**Issue by Aohzan** [amelchio/pysonos#51]
Hello,

All my playlists are detected as `Unhandled favorite` (Sonos playlists, Spotify...)
If I add radio stations, albums, artists... to My Sonos, they appears successfully on my Home Assistant.

Have you an idea what's the problem is ?
Thank you

**Comment by amelchio** [comment]
Including the error message could help.

**Comment by Aohzan** [comment]
sorry, I put it in hass forum but forgot here :
`2020-04-26 15:13:35 ERROR (Thread-35) [homeassistant.components.sonos.media_player] Unhandled favorite 'Indémodable': Unknown UPnP class: object.container.playlistContainer#playlistItem`

If it can help you, what I get using SoCo get_sonos_favorites() function:
```
<stdin>:1: UserWarning: Call to deprecated function get_sonos_favorites. Will be removed in version 0.15. Use soco.music_library.get_sonos_favorites instead.
/home/matthieu/.local/lib/python3.8/site-packages/soco/utils.py:164: UserWarning: The output type of this method will probably change in the future to use SoCo data structures
  return deprecated_fn(*args, **kwargs)
{'total': '5', 'returned': 5, 'favorites': [{'title': 'Indémodable', 'uri': 'x-rincon-cpcontainer:0006002cspotify%3aplaylist%3a37i9dQZF1DWST4Eo6XmJX0?sid=9&flags=44&sn=14', 'meta': '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="0006002cspotify%3aplaylist%3a37i9dQZF1DWST4Eo6XmJX0" parentID="spotify%3aview%3ayour-heavy-rotation%5b0%5d" restricted="true"><dc:title>Indémodable</dc:title><upnp:class>object.container.playlistContainer#playlistItem</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON2311_X_#Svc2311-0-Token</desc></item></DIDL-Lite>'}, {'title': 'Jazz Collection', 'uri': 'x-rincon-cpcontainer:1004206clibrary%2falbums%2f73418766-0d89-45f4-9eda-3398fb29e32c%2f%23library_tracks?sid=201&flags=8300&sn=17', 'meta': '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="1004206clibrary%2falbums%2f73418766-0d89-45f4-9eda-3398fb29e32c%2f%23library_tracks" parentID="100d2064library%2falbums%2f%23library_albums" restricted="true"><dc:title>The Perfect Jazz Collection - 25 Original Albums</dc:title><upnp:class>object.container.album.musicAlbum</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON51463_X_#Svc51463-0-Token</desc></item></DIDL-Lite>'}, {'title': 'Rire & Chansons NOUVELLE GENERATION', 'uri': 'x-sonosapi-stream:27090?sid=303&flags=8224&sn=19', 'meta': '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="1009202027090" parentID="00020000station:rire" restricted="true"><dc:title>Rire &amp; Chansons NOUVELLE GENERATION</dc:title><upnp:class>object.item.audioItem.audioBroadcast</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON77575_</desc></item></DIDL-Lite>'}, {'title': 'Rire et Chansons', 'uri': 'x-sonosapi-stream:6666?sid=303&flags=8224&sn=19', 'meta': '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="100920206666" parentID="00020000station:rire" restricted="true"><dc:title>Rire et Chansons</dc:title><upnp:class>object.item.audioItem.audioBroadcast</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON77575_</desc></item></DIDL-Lite>'}, {'title': 'RTL', 'uri': 'x-sonosapi-stream:4835?sid=303&flags=8224&sn=19', 'meta': '<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="100920204835" parentID="00020000station:rtl" restricted="true"><dc:title>RTL</dc:title><upnp:class>object.item.audioItem.audioBroadcast</upnp:class><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">SA_RINCON77575_</desc></item></DIDL-Lite>'}]}
```

**Comment by amelchio** [comment]
Interesting, my Spotify playlist favorites are `<upnp:class>object.container.playlistContainer</upnp:class>`.

Can you tell me the version of everything that you are running and every step you take to create such a favorite?

**Comment by Aohzan** [comment]
I have the latest update of Home Assistant 0.108.9, Sonos too 11.1. It doesn't work since 1 or 2 weeks I think.
I go on the Sonos app on Android or iOS, go in Spotify by example, long press on a playlist and "Add the playlist to My Sonos" (I translate myself from french, maybe not the exact sentence in english). 
If I try "Add to a Sonos playlist" and create a new playlist, I don't have an new error in logs, I see that hass refresh playlists, but I don't see the new playlist in source list

**Comment by Aohzan** [comment]
I just try to add a playlist through Windows application and it works, for the same playlists

**Comment by amelchio** [comment]
Thank you, now I managed to create such an entry.

**Comment by amelchio** [comment]
I am working on something else, will make a fixed release when both things are ready.

**Comment by Aohzan** [comment]
thank you, it's not hurry since I added the playlist with the Windows app :)

**Comment by amelchio** [comment]
Fixed in v0.0.27. I will soon update Home Assistant to use this version.
